packages/components/src/button/index.js

```diff
@@ -124,17 +132,33 @@ export function Button( props, ref ) {
 	);
 
 	if ( ! shouldShowTooltip ) {
-		return element;
+		return (
+			<>
+				{ element }
+				{ describedBy && (
+					<VisuallyHidden>
+						<span id={ descriptionId }>{ describedBy }</span>
+					</VisuallyHidden>
+				) }
+			</>
+		);
```

Comment:
It would probably be valid HTML, but that doesn't mean it would be valid a11y. Putting it inside the `<button>`/`<a>` would make the whole thing be considered part of the button label. I don't think that would be correct/desirable, but I could be mistaken.

Reply:
Yeah, the description shouldn't be inside the element being described. 

Potentially we could find another workaround by adding in actual button text as we did with the global inserter.

But the visually hidden elements are absolutely positioned; I wouldn't expect them to occupy any space at all in the canvas. Could that spacing be caused by some other issue?

Reply:
> Could that spacing be caused by some other issue?

We have some styles targetting last-child and first-child CSS pseudo elements causing these issues. So these addition conflicts with these styles. I'm considering exploring changes to these styles but with no guarantee of success.

Reply:
Oh I see, the problem is not `VisuallyHidden` but the pseudo-element we use to render the text label. In that case, we could definitely explore a similar approach to the global inserter: render actual text in the button instead of the pseudo-element.